osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DeviceInstallationServiceTest.java

```diff
@@ -0,0 +1,276 @@
+	
+	@InjectMocks
+	private DeviceInstallationService deviceInstallationService;
+
+	@Captor
+	private ArgumentCaptor<ResponseMessage> argumentResM;
+
+	@Captor
+	private ArgumentCaptor<RequestMessage> argumentReqM;
+
+	@Captor
+	private ArgumentCaptor<String> argumentStringOne;
+
+	@Captor
+	private ArgumentCaptor<String> argumentStringTwo;
+
+	@Captor
+	private ArgumentCaptor<Integer> argumentInt;
+	
+	@Test
+	public void testGetStatusTestDeviceTypeIsNotLmdType() throws FunctionalException {
+		final Device device = Mockito.mock(Device.class);
+		when(device.getIpAddress()).thenReturn("testIp");
+		when(this.deviceDomainService.searchActiveDevice("testDevice", ComponentType.DOMAIN_CORE)).thenReturn(device);
+		
+		this.deviceInstallationService.getStatus("testOrganisation", "testDevice", "testUid", "testMessageType", 1);
+
+		verify(this.osgpCoreRequestMessageSender).send(this.argumentReqM.capture(), this.argumentStringOne.capture(),
+				this.argumentInt.capture(), this.argumentStringTwo.capture());
+
+		assertThat(this.argumentReqM.getValue()).usingRecursiveComparison().isEqualTo(new RequestMessage(
+				"testUid",
+				"testOrganisation",
+				"testDevice", null));
+		assertThat(this.argumentStringOne.getValue()).isEqualTo("testMessageType");
+		assertThat(this.argumentInt.getValue()).isEqualTo(1);
+		assertThat(this.argumentStringTwo.getValue()).isEqualTo("testIp");
+	}
+	
+	@Test
+	public void testGetStatusTestDeviceTypeIsLmdType() throws FunctionalException {
+		final Device device = Mockito.mock(Device.class);
+		when(device.getDeviceType()).thenReturn("LMD");
+		when(device.getIpAddress()).thenReturn("testIp");
+		when(this.deviceDomainService.searchActiveDevice("testDevice", ComponentType.DOMAIN_CORE)).thenReturn(device);
+
+		this.deviceInstallationService.getStatus("testOrganisation", "testDevice", "testUid", "testMessageType", 1);
+
+		verify(this.osgpCoreRequestMessageSender).send(this.argumentReqM.capture(), this.argumentStringOne.capture(),
+				this.argumentInt.capture(), this.argumentStringTwo.capture());
+
+		assertThat(this.argumentReqM.getValue()).usingRecursiveComparison().isEqualTo(new RequestMessage(
+				"testUid",
+				"testOrganisation",
+				"testDevice", null));
+		assertThat(this.argumentStringOne.getValue()).isEqualTo("GET_LIGHT_SENSOR_STATUS");
+		assertThat(this.argumentInt.getValue()).isEqualTo(1);
+		assertThat(this.argumentStringTwo.getValue()).isEqualTo("testIp");
+	}
+
+	@Test
+	public void testHandleGetStatusResponseNotOk() {
+		this.deviceInstallationService.handleGetStatusResponse(null, new CorrelationIds("testOrganisation", "testDevice", "testUid"),
+				"testMessageType", 1, ResponseMessageResultType.NOT_OK, null);
+
+		verify(this.webServiceResponseMessageSender).send(this.argumentResM.capture());
+		assertThat(this.argumentResM.getValue()).usingRecursiveComparison().isEqualTo(ResponseMessage.newResponseMessageBuilder()
+				.withIds(new CorrelationIds("testOrganisation", "testDevice", "testUid"))
```

Comment:
Since
new CorrelationIds("testOrganisation", "testDevice", "testUid")
is used in other tests, refactor it to a constant